home/management/commands/importgradedata.py

```diff
@@ -72,15 +74,28 @@ def parse_professor(self, name: str):
         if name is None or name == "":
             return None
 
-        names = name.strip().split(",")
-        lastname = names[0]
-        firstname = names[-1].strip().split()[0]
-        professors = Professor.unfiltered.filter(
-            Q(name__istartswith=firstname) & Q(name__iendswith=lastname)
-        )
-        if professors.exists():
+        # .first() is ok here because at most one record will be returned
+        alias = ProfessorAlias.objects.filter(alias=name.strip()).first()
+        if alias:
+            return alias.professor
+
+        professors = Professor.verified.filter(name=name.strip())
+        if professors.count() == 1:
             return professors.first()
 
+        similar_professors = Professor.find_similar(name.strip(), 70)
+        if professors.count() > 1 or len(similar_professors):
```

Comment:
for an array `a`, `bool(a) == bool(len(a))` holds, so this truthy evaluation on `len(similar_professors)` is very weird. Just do `or similar_professors`.